### PR TITLE
create CI dhall components for building and caching the test-executive and executing integration tests

### PR DESCRIPTION
**Pull request by 0x0I**
Also increase Buildkite agent cluster capacity to handle increased CI load and add artifact caching helper script to temporarily address #7745.

**Test:** Buildkite CI

Checklist:

- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

### CODE REVIEW

File: buildkite/src/Command/ExecuteIntegrationTest.dhall [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+let deployEnv = "DOCKER_DEPLOY_ENV"
+
+in
+
+{ step = \(testName : Text) -> \(dependsOn : List Command.TaggedKey.Type) ->
+    Command.build
+      Command.Config::{
+        commands = [
+            Cmd.run (
+              "if [ ! -f ${deployEnv} ]; then " ++
```
Comment (0x0I) [inline buildkite/src/Command/ExecuteIntegrationTest.dhall:18]: created https://github.com/MinaProtocol/mina/issues/7745 to address DRY'ing this code up a bit

File: buildkite/src/Command/ExecuteIntegrationTest.dhall [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+            -- Build test executive binary and cache
+            -- TODO: cache using `cacheThrough` method (see: https://github.com/MinaProtocol/mina/blob/develop/buildkite/src/Jobs/Lint/ValidationService.dhall#L25)
+            Cmd.run "chmod -R 777 src/app/test_executive",
+            Cmd.runInDocker
+              Cmd.Docker::{
+                image = (../Constants/ContainerImages.dhall).codaToolchain,
+                extraEnv = [ "DUNE_PROFILE=testnet_postake_medium_curves" ]
+              }
+              "dune build --verbose --profile=\\\${DUNE_PROFILE} src/app/test_executive/test_executive.exe",
```
Comment (0x0I) [inline buildkite/src/Command/ExecuteIntegrationTest.dhall:31]: Make sense to create a `Makefile` directive for building the test-executive? Something like `make test-executive`...

**Review by ghost-not-in-the-shell — APPROVED** [review]

File: buildkite/src/Jobs/Test/TestnetIntegrationTests.dhall [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+let dependsOn = [
+    { name = "MinaArtifact", key = "mina-docker-image" }
+]
+
+in Pipeline.build Pipeline.Config::{
+  spec =
+    JobSpec::{
+    dirtyWhen = [
+        S.strictlyStart (S.contains "src"),
+        S.strictlyStart (S.contains "src/app/test-executive"),
```
Comment (0x0I) [inline buildkite/src/Jobs/Test/TestnetIntegrationTests.dhall:17]: remove

File: buildkite/scripts/build-test-executive.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
```
Comment (0x0I) [inline buildkite/scripts/build-test-executive.sh:1]: convert to Makefile directive?

File: automation/terraform/infrastructure/us-east1.tf [hunk 5]
```diff
@@ -222,21 +222,21 @@ resource "kubernetes_storage_class" "east1_standard" {
 ## Monitoring 
 
 provider helm {
-  alias = "helm_east1"
+  alias = "helm_east"
```
Comment (0x0I) [inline automation/terraform/infrastructure/us-east1.tf:225]: Legacy setting based on existing cluster terraform state file. Should replace w/ `east1` once it's ok to completely wipe and refresh the state.

File: automation/terraform/modules/kubernetes/buildkite-agent/helm.tf [hunk 6]
```diff
@@ -128,10 +131,54 @@ locals {
...
+
+        set -o pipefail
+
+        if [[ $1 ]]; then
+          export BUILDKITE_ARTIFACT_UPLOAD_DESTINATION="gs://buildkite_k8s/coda/shared/$${BUILDKITE_JOB_ID}"
+          FILE="$1"
+          DOWNLOAD_CMD="buildkite-agent artifact download --build $${BUILDKITE_BUILD_ID} --include-retried-jobs"
+
+          while [[ "$#" -gt 0 ]]; do case $1 in
+            --upload) UPLOAD_PATH="true"; shift;;
```
Comment (0x0I) [inline automation/terraform/modules/kubernetes/buildkite-agent/helm.tf:145]: replace envar => `UPLOAD`

**Review by lk86 — APPROVED** [review]
Looks good aside from the existing comments, a makefile directive definitely seems worthwhile

**Review by ghost-not-in-the-shell — APPROVED** [review]

File: buildkite/scripts/build-test-executive.sh (hunk 4 above)
Comment (0x0I) [reply buildkite/scripts/build-test-executive.sh:1]: Will update in subsequent PR (a part of BK cleanup/refactors)

**Pull request merged**